src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -111,8 +111,26 @@ public override BoundNode VisitNegatedPattern(BoundNegatedPattern node)
 
         public override BoundNode VisitBinaryPattern(BoundBinaryPattern node)
         {
-            Visit(node.Left);
-            Visit(node.Right);
+            // Users (such as ourselves) can have many, many nested binary patterns. To avoid crashing, do left recursion manually.
```

Comment:
>// Users (such as ourselves) can have many, many nested binary patterns. To avoid crashing, do left recursion manually. [](http://example.com/codeflow?start=12&length=119)

Should we add a similar override to `DebugVerifier`?